src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (arguments.length !== 0 && separator !== undefined) {
+      let resultStr = '';
+
+      for (let i = 0; i < this.length; i++) {
+        const item = (this[i] === undefined || this[i] === null)
+          ? '' : this[i];
+
+        if (i === this.length - 1) {
+          resultStr += item;
+        } else {
+          resultStr += item + `` + separator;
+        }
```

Comment:
```suggestion
        if (i !== this.length - 1) {
          resultStr += separator;
        }
```
in my variant of code you don't need else statement, just add separator to the resultStr, you should also replace === with !==